lib/src/layout/scaffold.dart

```diff
@@ -1,117 +1,303 @@
+                            Border(bottom: BorderSide(color: dividerColor)),
+                        borderRadius: widget.titleBar?.decoration?.borderRadius,
+                        boxShadow: widget.titleBar?.decoration?.boxShadow,
+                        gradient: widget.titleBar?.decoration?.gradient,
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+
+            // Sidebar resizer
+            if (widget.sidebar?.isResizable ?? false)
+              Positioned(
+                left: _sidebarWidth - 4,
+                width: 7,
+                height: height,
+                child: GestureDetector(
+                  behavior: HitTestBehavior.opaque,
+                  onHorizontalDragUpdate: (details) {
+                    setState(() {
+                      _sidebarWidth = math.max(
+                        widget.sidebar!.minWidth,
+                        math.min(
+                          math.min(widget.sidebar!.maxWidth!, width),
+                          _sidebarWidth + details.delta.dx,
+                        ),
+                      );
+                    });
+                  },
+                  child: MouseRegion(
+                    cursor: SystemMouseCursors.resizeColumn,
+                    child: Align(
+                      alignment: Alignment.center,
+                      child: VerticalDivider(
+                        thickness: 1,
+                        width: 1,
+                        color: dividerColor,
+                      ),
                     ),
-                    view2: body,
-                    viewMode: SplitViewMode.Horizontal,
-                  );
-                } else {
-                  return body;
-                }
-              },
-            )
-          : body,
+                  ),
+                ),
+              ),
+          ],
+        );
+
+        return ValueListenableBuilder<Map<Key, double>>(
+          child: layout,
+          valueListenable: _valueNotifier,
+          builder: (_, panes, child) {
+            double sum = panes.values.fold(0.0, (prev, curr) => prev + curr);
+            double _remainingWidth = width - (_sidebarWidth + sum);
+
+            return ScaffoldScope(
+              child: child!,
+              constraints: constraints,
+              valueNotifier: _valueNotifier,
+              contentAreaWidth: math.max(_minContentAreaWidth, _remainingWidth),
+            );
+          },
+        );
+      },
     );
   }
 }
+
+class ScaffoldScope extends InheritedWidget {
+  const ScaffoldScope({
+    Key? key,
+    required this.constraints,
+    required this.contentAreaWidth,
+    required Widget child,
+    required this.valueNotifier,
+  }) : super(key: key, child: child);
+
+  final BoxConstraints constraints;
+
+  final double contentAreaWidth;
+
+  final ResizablePaneNotifier valueNotifier;
+
+  static ScaffoldScope of(BuildContext context) {
```

Comment:
There should also be a `maybeOf(BuildContext)` method that returns a nullable `ScaffoldScope?`